### ISSUE TITLE
Issue with using `*ngIf` and `*ngSwitch` directives in Modal Page for iOS

### ISSUE THREAD

**Issue by tsonevn** [NativeScript/nativescript-angular#793]
Both `*ngIf` and `*ngSwitch` directives will not work as expected for iOS, while using it in Modal Page.
Reported by @piotrilski

> It seems to be affecting usage of *ngIf and *ngSwitch on angular as well. I made a fork of sdk-examples-ng repo here: https://github.com/piotrilski/nativescript-sdk-examples-ng/tree/pilski/modal-issue
Changes I made: https://github.com/piotrilski/nativescript-sdk-examples-ng/pull/1/files
To see what's going on simply run it and go to ModalPage -> Modal Page Example and hit first EnterStartDate button to open a slightly modified modal. It behaves exactly like in

**Comment by piotrilski** [comment]
There's a workaround described by @kssfilo here: https://github.com/NativeScript/NativeScript/issues/4151#issuecomment-300712894

TL;DR - importing Page and using this method:
```page.nativeView.setNeedsLayout()```
once the value binded to ngIf/ngSwitch changed sorted out the issue.
